src/index.js

```diff
@@ -1,44 +1,110 @@
 
-const listTask = document.querySelector('.list');
+const list = [];
 
-window.addEventListener('load', () => {
-  list.sort((a, b) => a.index - b.index);
-
-  for (let i = 0; i < list.length; i += 1) {
-    listTask.innerHTML += `
-         <li class="d-flex-row">
-             <div class="list_task d-flex-row">
-                <input type="checkbox" name="task" class="checkbox">
-                <span>${list[i].description}</span>
-             </div>
-             <img src=${more} alt="more icon">
-            </li>
-            <li class="break_line_list"><hr></li>
-         `;
+class Task {
+  constructor(value) {
+    this.description = value;
+    this.completed = false;
+    this.index = list.length + 1;
   }
+}
+
+// Add to array and Store it in localestorage
+
+const form = document.querySelector('#list_input');
+const input = document.querySelector('#input');
+
+// Display //
+
+const addBtn = document.querySelector('.add_btn');
+const listTask = document.querySelector('.list');
+
+function SaveAndDisplay() {
+  const newTask = new Task(input.value);
+  add(list, newTask);
+
+  localStorage.setItem('ToDoList', JSON.stringify(list));
+
+  // add <li>
+  const li = document.createElement('li');
+  li.classList.add('li', 'd-flex-row');
+  listTask.appendChild(li);
+
+  // add <div>
+  const div = document.createElement('div');
+  div.classList.add('list_task', 'd-flex-row');
+  li.appendChild(div);
+
+  // add <input>
+  const inputTag = document.createElement('input');
+  inputTag.type = 'checkbox';
+  inputTag.classList.add('checkbox');
+  div.appendChild(inputTag);
+
+  // add <span>
+  const span = document.createElement('span');
+  span.innerText = input.value;
+  div.appendChild(span);
+
+  // add <img>
+  const moreBtn = document.createElement('img');
+  moreBtn.classList.add('more_btn');
+  moreBtn.id = newTask.index;
+  moreBtn.src = more;
+  moreBtn.alt = 'more icon';
+  li.appendChild(moreBtn);
+
+  // Display //
+
+  input.value = '';
+
+  // Edit Task
+  span.addEventListener('dblclick', (e) => {
+    UpdateTask(e);
+    const taskInput = document.querySelector('.edit');
+    taskInput.addEventListener('keypress', (e) => {
+      if (e.keyCode === 13) {
+        newTask.description = taskInput.value;
+        localStorage.setItem('ToDoList', JSON.stringify(list));
+        saveUpdatedTask(taskInput.value, div, taskInput);
+      }
+    });
+  });
+
+  // remove Task
+
+  moreBtn.addEventListener('click', (e) => {
+    const idNumer = (parseInt(e.target.id, 10) - 1);
+    e.target.parentNode.remove();
+    removeTask(list, idNumer);
+
+    // Update Index
+
+    list.forEach((task, i) => {
+      task.index = i + 1;
+    });
+
+    // Update id
+
+    const moreBtns = document.querySelectorAll('.more_btn');
+    moreBtns.forEach((btn) => {
+      if (btn.id > e.target.id) {
+        btn.id -= 1;
+      }
+    });
+
+    localStorage.setItem('ToDoList', JSON.stringify(list));
+  });
+}
```

Comment:
- [ ] Let's talk about a few issues we have with this function:
  
  1. This function is too long and handles many responsibilities, this is not a good code practice. Just like in the project requirements, adding a new task, deleting a task, and editing a task should all be handled by separate functions. See below.
    
  ![image](https://user-images.githubusercontent.com/45002654/149360146-5c9a0044-9a50-414f-a39a-fa0fc7f8b6cd.png)
  
  2. Functions should be single-responsibility functions, this makes it easier to maintain code in case one functionality fails to work during testing.
  
  ### Some suggestions
  - Write separate functions for adding, editing, and deleting tasks
  - Write a separate function for displaying tasks on the screen. This function should accept an array of tasks and use it to populate an HTML list and display it on the UI.
  
  Example:
  ```JavaScript
  // First check if there are tasks in the localStorage, if not use an empty array.
  const list = JSON.parse(localStorage.getItem('ToDoList')) || [];
  
   // implement the displayTasks function
  const displayTasks = (tasks) => {
    // add logic here
  };
  
  // call the displayTask function every time you add, delete, or edit a task
  const addTask = () => {
    // add logic for adding a task
    // call the displayTask function
    displayTasks(list);
  };
  
  const editTask = () => {
    // add logic for editing a task
    // call the displayTask function
    displayTasks(list);
  };
  
  const deleteTask = () => {
    // add logic for deleting a task
    // call the displayTask function
    displayTasks(list);
  };
  
  // For a good user experience you can add a `window.onload` event listener to display tasks every time the browser window is loaded or reloaded
  window.onload = () => {
    displayTasks(list);
  };
  ```
  
  Please refactor accordingly.